apps/scan/frontend/src/app_root.tsx

```diff
@@ -316,11 +308,17 @@ export function AppRoot({
     );
   }
 
-  if (isCastVoteRecordSyncRequiredScreenUp) {
+  if (
+    usbDrive.doesUsbDriveRequireCastVoteRecordSync ||
+    // This ensures that we don't immediately transition away from the CVR sync success message.
+    // We can't rely on doesUsbDriveRequireCastVoteRecordSync because it becomes false as soon as
+    // the sync completes.
+    shouldStayOnCastVoteRecordSyncRequiredScreen
+  ) {
```

Comment:
The fix for bug 3 in the PR description

> The only way to currently clear the CVR sync required screen is to click the screen's sync button. But other actions can also render a sync to USB drive unnecessary. One such action is switching from test mode to official mode, since switching mode clears scanned ballots. These other actions should also clear the CVR sync required screen.

I previously wasn't using `usbDrive.doesUsbDriveRequireCastVoteRecordSync` directly because I needed the CVR sync required screen to stay open after sync completion to display the success message. But my solution for that ended up introducing the mentioned bug. Now, I'm using `usbDrive.doesUsbDriveRequireCastVoteRecordSync` directly, with a separate solution for the success message - best of both worlds